src/createMockMiddleware.ts

```diff
@@ -38,43 +46,50 @@ export default function getMockMiddleware(mockDir?: string, options?: MockOption
     });
     watcher.on('all', (event, file) => {
       debug(`[${event}] ${file}, reload mock data`);
-      mockData = getConfig();
+      mockDataPromise = getConfig();
       if (!errors.length) {
         signale.success(`Mock file parse success`);
       }
     });
   }
 
-  function getConfig() {
+  async function getConfig(): Promise<Array<MockConfig>> {
     // Clear errors
     errors.splice(0, errors.length);
 
     cleanRequireCache();
     let ret = {};
     const mockFiles = glob
-      .sync('**/*.[jt]s', {
+      .sync('**/*.@(cjs|mjs|js|ts)', {
         cwd: absMockPath,
         ignore: (options || {}).ignore || [],
       })
       .map((p) => join(absMockPath, p));
     debug(`load mock data from ${absMockPath}, including files ${JSON.stringify(mockFiles)}`);
     try {
-      ret = mockFiles.reduce((memo, mockFile) => {
-        const m = require(mockFile); // eslint-disable-line
-        memo = {
-          ...memo,
-          ...(m.default || m),
-        };
-        return memo;
-      }, {});
-    } catch (e) {
-      errors.push(e);
-      signale.error(`Mock file parse failed: ${e.message}`);
+      ret = (
+        await Promise.all(
+          mockFiles.flatMap((mockFile) => {
+            if (USE_ESM) {
```

Comment:
this is getting a bit hairy, how about
```suggestion
                       const module = await (USE_ESM ? import(`${mockFile}?cache=${Math.random()}`) : Promise.resolve(require(mockFile)))
            return module.default || module;

```

Reply:
Ah yes that's better!

No need for the `Promise.resolve`, you can `await` normal stuff as well :)
```
const module = await (USE_ESM
              ? import(`${mockFile}?cache=${Math.random()}`)
              : require(mockFile));
```